pkg/v1/providers/ytt/02_addons/csi/csi_addon_data.lib.yaml

```diff
@@ -1,6 +1,27 @@
 #@ load("@ytt:data", "data")
+#@ load("@ytt:regexp", "regexp")
+#@ load("@ytt:assert", "assert")
 #@ load("/lib/helpers.star", "get_bom_data_for_tkr_name", "get_image_repo_for_component", "get_no_proxy")
 
+
+#! Set topologyFeatureFlag to be true for TKR 1.22 and newer
+#@ def enableTopologyFeature():
+#@ bomData = get_bom_data_for_tkr_name()
+#@ if regexp.match("^v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$", bomData.release.version) != True:
```

Comment:
(blocking comment): If we really need to use this regex here we'll need to add unit tests for templating that cover it along with documentation. Alternatively try to use [version](https://carvel.dev/ytt/docs/latest/lang-ref-ytt-version/) module

Reply:
Removed regex mathing. Did some best effort validation for TKR versions.